### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by thatblindgeye**
## Because

Currently we have a contributing guide in our org `.github` directory as well as our web app repo. We should only have one source of truth, so this helps split out repo specific content only to its repo while also linking to the "general" contributing content.

## Issue

Closes #XXXXX

## Pull Request Requirements

-   [ ] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [ ] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [ ] The `Because` section summarizes the reason for this PR
-   [ ] The `This PR` section has a bullet point list describing the changes in this PR
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [ ] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: CONTRIBUTING.md
Comment (thatblindgeye) [inline CONTRIBUTING.md:27]: This depends on the PR I have up adding the dev dep of markdownlint getting merged. If it doesn't then we can revert this to the instructions on manually installing the dependency itself.

File: CONTRIBUTING.md
Comment (MaoShizhong) [inline CONTRIBUTING.md:33]:
```suggestion
With either of these two methods, keep in mind that not all issues that get flagged will have an autofix available. Some rules require fixes that are more dependent on context and cannot - and should not - be automatically fixed, such as our custom rule `TOP001` for descriptive link text.
```

File: CONTRIBUTING.md
Comment (MaoShizhong) [inline CONTRIBUTING.md:27]: Does running them without the file path lint/fix *all* files on your end?
According to the [markdownlint-cli2 docs](https://github.com/DavidAnson/markdownlint-cli2?tab=readme-ov-file#command-line), that would only occur if you specify `.` or `**` as the arg, and passing no arg just results in that help output showing (and no lint/fix action applied).
That's how it works on my end, and is definitely preferable behaviour of course.

File: CONTRIBUTING.md
Comment (MaoShizhong) [inline CONTRIBUTING.md:16]:
```suggestion
1. If you're more experienced with contributing or need to edit more than 1 file, you can follow our instructions on [setting up a local clone](https://github.com/TheOdinProject/.github/blob/ac0adb6bd151dca53798776783c5604d31e3389e/CONTRIBUTING.md#setting-up-your-local-clone) from our general contributing guide. You should also read the sections that follow on how to open a PR.
```

File: CONTRIBUTING.md
Comment (MaoShizhong) [inline CONTRIBUTING.md:26]:
```suggestion
1. Install the [Markdownlint VSCode Plugin](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint). This plugin will automatically pick up our main configuration and flag issues with a squiggly underline. Note that the plugin will not currently flag issues with the structure of lessons or projects.
```

> Note that the plugin will not currently flag issues with the structure of lessons or projects.

I feel this could be a bit clearer as to what "the structure of lessons or projects" means? e.g. also mentioning it's the TOP004 rule, and/or that it's because the extension doesn't use the rules from the `(lesson|project).-*` configs

File: CONTRIBUTING.md
Comment (MaoShizhong) [inline CONTRIBUTING.md:27]: > which we cover in our Foundations path.

Worth linking to the Node installations lesson here or not?

File: CONTRIBUTING.md
Comment (MaoShizhong) [inline CONTRIBUTING.md:20]:
```suggestion
Before submitting a PR for any lesson, you must also use our [Lesson Preview Tool](https://www.theodinproject.com/lessons/preview) to ensure the lesson Markdown is correctly formatted and rendering properly.
```
Just a suggestion, your call. Though at least with the original, we wouldn't need the second indefinite article after "or".

File: CONTRIBUTING.md
Comment (thatblindgeye) [reply CONTRIBUTING.md:27]: I was considering it, but ended up not just to avoid having users still early on skipping to that lesson. May not be a huge deal I suppose, but that was at least why I didn't include a link.

Comment (thatblindgeye) [reply CONTRIBUTING.md:27]: Oop you're right, updating

Comment (MaoShizhong) [reply CONTRIBUTING.md]: That makes sense!

File: CONTRIBUTING.md
Comment (MaoShizhong) [inline CONTRIBUTING.md:7]:
```suggestion
## Table of Contents

```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Just noticed this nit, but otherwise the rest looks good to me. Let's wait for @ChargrilledChook's review, as I've probably overlooked some stuff.

**Review by ChargrilledChook — APPROVED** [review]

**Review by MaoShizhong — APPROVED** [review]

**Comment by MaoShizhong** [comment]
Just thought as well, @thatblindgeye, would you also want to update this repo's [PR template](https://github.com/TheOdinProject/curriculum/blob/main/.github/PULL_REQUEST_TEMPLATE.md?plain=1) checklist to include this new Contributing Guide?

**Comment by thatblindgeye** [comment]
@MaoShizhong yep! I'll try to go around each repo to find where we're linking to contributing guides and see what needs to be updated

**Comment by MaoShizhong** [comment]
Awesome. I'm aware at least that in the curriculum repo, the "Using git in the real world" lesson also does as a first step somewhere.

File: CONTRIBUTING.md
Comment (Asartea) [inline CONTRIBUTING.md:3]: Is there a reason this is linking to a specific commit instead of the version on the main branch (ie https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)?

File: CONTRIBUTING.md
Comment (Asartea) [inline CONTRIBUTING.md:17]: Same as above

File: CONTRIBUTING.md
Comment (thatblindgeye) [reply CONTRIBUTING.md]: Yep!

I copied the url from the wrong tab 😆

**Pull request merged**
